### PR TITLE
Add `Alternative` instance for `Bijection`

### PR DESCRIPTION
**Pull request by ChShersh**
This is a first step towards supporting sum types. Not sure that I'm moving in the right direction... What I want is to have `Alternative` instance for `Bijection`.

### CODE REVIEW

File: src/Toml/Bi/Code.hs [hunk 1]
```diff
@@ -67,6 +86,7 @@ decode biToml text = do
     toml <- first ParseError (parse text)
     runReader (runExceptT $ biRead biToml) toml
 
--- | Convert object to textual representation.
+-- | Convert object to textual representation. Outputs empty string if for some
+-- reason not able to convert to 'Text'.
```
Comment (ChShersh) [inline src/Toml/Bi/Code.hs]: Oops, this comment is wrong. It doesn't output empty string.

File: src/Toml/Bi/Code.hs [hunk 2]
```diff
@@ -31,15 +32,26 @@ import qualified Data.Text as Text
...
+    = TrivialError
+    | KeyNotFound Key  -- ^ No such key
     | TableNotFound Key  -- ^ No such table
     | TypeMismatch Key Text TValue  -- ^ Expected type vs actual type
     | ParseError ParseException  -- ^ Exception during parsing
     deriving (Eq, Show)  -- TODO: manual pretty show instances
 
+instance Semigroup DecodeException where
+    TrivialError <> e = e
+    e <> TrivialError = e
```
Comment (vrom911) [inline src/Toml/Bi/Code.hs]: I guess this case is not needed

File: src/Toml/Bi/Code.hs [hunk 3]
```diff
@@ -53,9 +65,16 @@ prettyException = \case
 -- This is @r@ type variable in 'Bijection' data type.
 type Env = ExceptT DecodeException (Reader TOML)
 
--- | Mutable context for 'Toml' conversion.
+{- | Mutable context for 'Toml' conversion.
 -- This is @w@ type variable in 'Bijection' data type.
```
Comment (vrom911) [inline src/Toml/Bi/Code.hs]: You can remove `--` from this line

File: src/Toml/Bi/Code.hs [hunk 4]
```diff
@@ -53,9 +65,16 @@ prettyException = \case
...
-type St = State TOML
+
+@
+MaybeT (State TOML) a
+    = State TOML (Maybe a)
+    = TOML -> (Maybe a, TOML)
```
Comment (vrom911) [inline src/Toml/Bi/Code.hs]: Cool note!

File: src/Toml/Bi/Code.hs (hunk 2 above)
Comment (ChShersh) [reply src/Toml/Bi/Code.hs]: @vrom911 You're right. And removing this case is actually a good improvement, because it will make implementation lazier.

**Pull request merged**